cmd/stats/stats.go

```diff
@@ -2,28 +2,223 @@ package stats
 
 import (
 	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"time"
+
+	"strings"
+
+	. "github.com/NicksPatties/sweet/db"
 	"github.com/spf13/cobra"
 )
 
-var Command = &cobra.Command{
+var Cmd = &cobra.Command{
 	Use:   "stats",
 	Short: "Print statistics about typing exercises",
 	Run: func(cmd *cobra.Command, args []string) {
-		since, _ := cmd.Flags().GetString("since")
-		printStats(since)
+		q, err := argsToQuery(cmd, time.Now())
+		if err != nil {
+			log.Fatal()
+			os.Exit(-1)
+		}
+		reps, err := queryToReps(q)
+		if err != nil {
+			log.Fatal()
+			os.Exit(-1)
+		}
+		cols := argsToColumnFilter(cmd)
+		printStats(reps, cols)
 	},
 }
 
-func printStats(since string) {
-	if since != "" {
-		fmt.Printf("Printing stats since %s\n", since)
+// Returns the correct date depending on the start argument.
+// The time returned is at midnight of the date specified by `start`,
+// unless an hour query is used, in which case it will return the time
+// specified by
```

Comment:
```
// Returns the date depending on the provided argument.
// If this is an argument for the `start` or `since` flag, then return the date
// corresponding to midnight of the provided argument, otherwise return
// the date at one nanosecond before midnight of the next day.
// This function supports both `YYYY-MM-DD` arguments as well as `N[HDWMY]` arguments,
// where `N` is the number of hours, days, weeks, months, or years before `now`.
// The `now` argument is used primarily for testing, and is typically assigned to
// `time.Now()` in normal usage.